apps/web/src/app/(private)/_data-access/index.ts

```diff
@@ -54,20 +54,28 @@ export const getDocumentByUuid = cache(
     documentUuid: string
     commit: Commit
   }) => {
-    const workspace = await findWorkspaceFromCommit(commit)
```

Comment:
This pattern for finding the workspace on commit can be solved in a more generic way when returning from a repository `.find` method. Instead of returning just the model we return the model with their relations. At least some of then. 

What we're doing now is doing 1 more SQL query to get again the Workspace when we already got it to build the scope for that repository. I think this is doable. I might try in another PR.